webpack/loaders/manifest-loader.ts

```diff
@@ -1,26 +1,27 @@
-'use strict';
+/* eslint-disable no-invalid-this */
+import fs from 'fs';
 
-const fs = require('fs');
+import webpack from 'webpack';
 
-module.exports = function(content, map, meta) {
+export default function(this: webpack.LoaderContext<never>, _content: string): string {
   this.cacheable(true);
```

Comment:
I tested this locally and it works, but it looks like you're changing the parameters here from (content) to (this, content)? Can you help me understand why this works? Is the `this` parameter the thing you're calling `this.cacheable` on (and if so, should it be renamed to context)?

Reply:
See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#specifying-the-type-of-this-for-functions

> By default the type of `this` inside a function is `any`. Starting with TypeScript 2.0, you can provide an explicit `this` parameter. `this` parameters are fake parameters that come first in the parameter list of a function:

So it can't be renamed.

Reply:
Oh, thank you! I didn't know about this feature.